From d8f5eafe93ec628042b601353d884cbf24dbafd5 Mon Sep 17 00:00:00 2001
From: NaveenVenturi1203 <venturi.naveen@intel.com>
Date: Tue, 26 Nov 2024 08:32:45 +0000
Subject: [PATCH 1/2] Handle WindowInsets for Camera2 Settings Fragment

Issue Detailed: Camera Settings Fragment is not handling WindowInsets
leading to Preference Screens getting cropped and not displayed

IssueFixed: WindowInsets are fetched and preference screen container is
set after the status bar

Tracked-On: OAM-127731
Signed-off-by: NaveenVenturi1203 <venturi.naveen@intel.com>
---
 .../settings/CameraSettingsActivity.java       | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/src/com/android/camera/settings/CameraSettingsActivity.java b/src/com/android/camera/settings/CameraSettingsActivity.java
index 6c76e961c..235577a30 100644
--- a/src/com/android/camera/settings/CameraSettingsActivity.java
+++ b/src/com/android/camera/settings/CameraSettingsActivity.java
@@ -33,6 +33,13 @@ import android.preference.PreferenceGroup;
 import android.preference.PreferenceScreen;
 import androidx.fragment.app.FragmentActivity;
 import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.LayoutInflater;
+import androidx.annotation.NonNull;
+import androidx.core.view.ViewCompat;
+import androidx.core.view.WindowInsetsCompat;
+import androidx.core.graphics.Insets;
 
 import com.android.camera.FatalErrorHandler;
 import com.android.camera.FatalErrorHandlerImpl;
@@ -185,6 +192,17 @@ public class CameraSettingsActivity extends FragmentActivity {
             CameraAgentFactory.recycle(CameraAgentFactory.CameraApi.API_1);
         }
 
+        @Override
+        public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,Bundle savedInstanceState) {
+            View view = super.onCreateView(inflater,container,savedInstanceState);
+            ViewCompat.setOnApplyWindowInsetsListener(view,(v, insets) -> {
+                Insets systemInsets = insets.getInsets(WindowInsetsCompat.Type.systemBars());
+                v.setPadding(0,systemInsets.top,0,systemInsets.bottom);
+                return insets;
+            });
+            return view;
+	}
+
         @Override
         public void onResume() {
             super.onResume();
-- 
2.34.1

